### ISSUE TITLE
Inherited guards not being applied

### ISSUE THREAD

**Issue by wadedos** [nestjs/nest#879]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Controller guard not being applied when inherited. Only when a method on the child is defined with a guard, will the parents guard annotation apply.

## Expected behavior
Controller guard being applied when inherited

## Minimal reproduction of the problem with instructions
```typescript
import { Controller, Get, UseGuards, Injectable, CanActivate, ExecutionContext } from '@nestjs/common';

@Injectable()
export class TestGuard implements CanActivate {
  canActivate(
    context: ExecutionContext,
  ): boolean {
    console.log('Test Guard');
    return false;
  }
}

class ParentController {

  @Get('public')
  public public() {
    return 'parent - public';
  }

  @Get('protected')
  @UseGuards(TestGuard)
  public protected() {
    return 'parent - protected';
  }

}

@Controller('test')
export class ChildController extends ParentController {

}
```

How ever when you add the following function and guard annotations into the child controller, the parent guards finally get applied onto the /test/protected route.

```typescript
@Controller('test')
export class ChildController extends ParentController {
    @Get('direct')
    @UseGuards(TestGuard)
    public direct() {
        return 'child - protected';
    }
}

```

## What is the motivation / use case for changing the behavior?
N/A

## Environment

<pre><code>
[System Information]
OS Version     : macOS High Sierra
NodeJS Version : v10.6.0
NPM Version    : 6.1.0
[Nest Information]
common version : 5.0.0
core version   : 5.0.0
</code></pre>

**Comment by kamilmysliwiec** [comment]
Fixed in #907

**Comment by kamilmysliwiec** [comment]
5.2.0 is here!
